bpf/lbmap/lbmap.go

```diff
@@ -173,13 +225,95 @@ func LookupRevNat(key RevNatKey) (RevNatValue, error) {
 	return revnat.Convert(), nil
 }
 
+// gcd computes the gcd of two numbers.
+func gcd(x, y uint16) uint16 {
+	for y != 0 {
+		x, y = y, x%y
+	}
+	return x
+}
+
+// Generates a wrr sequence based on provided weights.
+func GenerateWrrSeq(weights []uint16) *RRSeqValue {
+	svcRRSeq := RRSeqValue{}
+
+	n := len(weights)
+	if n < 2 {
+		return nil
+	}
+
+	g := uint16(0)
+	for i := 0; i < n; i++ {
+		if weights[i] != 0 {
+			g = gcd(g, weights[i])
+		}
+	}
+
+	// This means all the weights are 0.
+	if g == 0 {
+		return nil
+	}
+
+	sum := uint16(0)
+	for i := range weights {
+		// Normalize the weights.
+		weights[i] = weights[i] / g
+		sum += weights[i]
+	}
+
+	// Check if Generated seq fits in our array.
+	if int(sum) > len(svcRRSeq.Idx) {
+		return nil
+	}
+
+	// Generate the Sequence.
+	i := uint16(0)
+	k := uint16(0)
+	for {
+		j := uint16(0)
+		for j < weights[k] {
+			svcRRSeq.Idx[i] = k
+			i++
+			j++
+		}
+		if i >= sum {
+			break
+		}
+		k++
+	}
+	svcRRSeq.Count = uint16(sum)
+	return &svcRRSeq
+}
+
+// Updates bpf map with the generated wrr sequence.
+func UpdateWrrSeq(fe ServiceKey, weights []uint16) error {
+	sum := uint16(0)
+	for i := 0; i < len(weights); i++ {
+		sum += weights[i]
+	}
+	if sum == 0 {
+		return nil
+	}
+	svcRRSeq := GenerateWrrSeq(weights)
```

Comment:
This becomes:
svcRREeq, err := GenerateWrrSeq(weights)
if err != nil {
   return fmt.Errorf("unable to generated weighted round robin seq for %+v with %+v: %s", fe, weights, err)
}